array.go

```diff
@@ -570,7 +570,7 @@ func (a *arrayObject) MemUsage(ctx *MemUsageContext) (uint64, error) {
 		return total, err
 	}
 
-	if len(a.values) > arrayLenThreshold {
+	if ctx.ArrayLenExceedsThreshold(len(a.values)) {
```

Comment:
Good question! Both here and where I check the mem usage don't need to know what the thresholds/limits are, all they should care about is if they stay within them. Since we are also the only ones consuming this it wouldn't be an issue to expose them, I feel uneasy allowing values to be mutable when they don't really need to.

For the `nil` issue we don't do any of those checks for anything else in the memory context as we rely on `NewMemUsageContext` to properly initiate it. We could add some validation to `NewMemUsageContext` if we are concerned about invalid values being passed in, I wouldn't mind at all!

Was there anything I did not take into account here ?

PS = I actually just noticed that `NewMemUsageContext` takes `vm *Runtime` as a first param but we never use it anywhere so I might clean that up.

Reply:
There's nothing stopping someone from instantiating `MemUsageContext` without the `New` helper func, so for completeness it would be good to do a nil check first.


As for the vm *Runtime param, I'm good to clean that up!